inference/enterprise/workflows/enterprise_blocks/sinks/opc_writer/v1.py

```diff
@@ -537,69 +537,69 @@ def _opc_connect_and_write_value(
             object_components = object_name.split("/")
             object_path = "/".join([f"{nsidx}:{comp}" for comp in object_components])
             node_path = f"0:Objects/{object_path}/{nsidx}:{variable_name}"
-            logging.debug(f"OPC Writer using hierarchical path: {node_path}")
+            logger.debug(f"OPC Writer using hierarchical path: {node_path}")
             var = client.nodes.root.get_child(node_path)
-            logging.debug(
+            logger.debug(
                 f"OPC Writer successfully found variable node using hierarchical path"
             )
         except BadNoMatch as exc:
-            logging.error(f"OPC Writer hierarchical path not found: {exc}")
+            logger.error(f"OPC Writer hierarchical path not found: {exc}")
             safe_disconnect(client)
             raise Exception(
                 f"WRONG OBJECT OR PROPERTY ERROR: Could not find node at hierarchical path '{node_path}'. Error: {exc}"
             )
         except Exception as exc:
-            logging.error(f"OPC Writer unhandled node lookup error: {type(exc)} {exc}")
+            logger.error(f"OPC Writer unhandled node lookup error: {type(exc)} {exc}")
             safe_disconnect(client)
             raise Exception(f"UNHANDLED ERROR: {type(exc)} {exc}")
 
     try:
-        logging.debug(
+        logger.debug(
             f"OPC Writer writing value '{value}' to variable with type '{value_type}'"
         )
-        # Use proper OPC UA type specification using VariantType enum
+        # Convert to primitive types before setting value
         if value_type in [BOOLEAN_KIND, "Boolean"]:
-            var.set_value(value, VariantType.Boolean)
+            var.set_value(bool(value), VariantType.Boolean)
         elif value_type == "Double":
-            var.set_value(value, VariantType.Double)
+            var.set_value(float(value), VariantType.Double)
         elif value_type in [FLOAT_KIND, "Float"]:
-            var.set_value(value, VariantType.Float)
+            var.set_value(float(value), VariantType.Float)
         elif value_type == "Int16":
-            var.set_value(value, VariantType.Int16)
+            var.set_value(int(value), VariantType.Int16)
```

Comment:
shuld maybe do bound check here also?

Reply:
Good point - the OPC server throws an error here but it's not obvious what the problem is. I updated.